quickwit/quickwit-indexing/src/actors/merge_executor.rs

```diff
@@ -105,7 +105,7 @@ impl Handler<MergeScratch> for MergeExecutor {
                     "Delete tasks can be applied only on one split."
                 );
                 assert_eq!(merge_scratch.tantivy_dirs.len(), 1);
-                let split_with_docs_to_delete = merge_op.splits.into_iter().next().unwrap();
+                let split_with_docs_to_delete = merge_op.splits.first().unwrap().clone();
```

Comment:
```suggestion
                let split_with_docs_to_delete = merge_op.splits[0].clone();
```

There is already an assert right above